common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -573,15 +573,16 @@ public UTF8String trim() {
    */
   public UTF8String trimAll() {
     int s = 0;
+    byte currentByte;
     // skip all of the whitespaces (<=0x20) in the left side
-    while (s < this.numBytes && Character.isWhitespace(getByte(s))) s++;
+    while (s < this.numBytes && 0 <= (currentByte = getByte(s)) && currentByte <= 0x20) s++;
```

Comment:
Seems like this would be a good time to pull this logic out into a shared method? The if-condition is getting a bit more complicated and hard to read with the temporary variable.

Reply:
Thank you for your suggestion but this issue seems to be regarded as a doc issue. So, I've decided not to change the code.